apps/teams-test-app/src/components/PagesAPIs.tsx

```diff
@@ -115,9 +131,15 @@ const SetCurrentFrame = (): React.ReactElement =>
           throw new Error('websiteUrl and contentUrl are required.');
         }
       },
-      submit: async input => {
-        pages.setCurrentFrame(input);
-        return 'called';
+      submit: {
+        withPromise: async input => {
+          pages.setCurrentFrame(input);
+          return 'called';
+        },
+        withCallback: (input, setResult) => {
+          setFrameContext(input);
+          setResult('called');
```

Comment:
you're probably going to hate me for this, but now seeing how many of these call `setResult` anyway, maybe we should add the option to return a string from here back :D Let's chat about this when we meet later today.

Reply:
It's interesting because there are a lot of calls that would also lead to importing `noHostSdk` if we add this back in, do you think we should modify `withCallback` to allow for both scenarios? One where we would want to return a string and the one where we don't?